fedimint-api/src/db/mod.rs

```diff
@@ -134,21 +134,36 @@ pub trait IDatabaseTransaction<'a>: 'a + Send {
 
 // TODO: use macro again
 #[doc = " A handle to a type-erased database implementation"]
-pub struct DatabaseTransaction<'a>(
-    Box<dyn IDatabaseTransaction<'a> + Send + 'a>,
-    ModuleDecoderRegistry,
-);
+pub struct CommitTracker {
+    is_committed: bool,
+}
+
+impl Drop for CommitTracker {
+    fn drop(&mut self) {
+        if !self.is_committed {
+            warn!("DatabaseTransaction has not called commit.");
+        }
+    }
+}
+
+#[doc = " A handle to a type-erased database implementation"]
+pub struct DatabaseTransaction<'a> {
+    tx: Box<dyn IDatabaseTransaction<'a> + Send + 'a>,
+    decoders: ModuleDecoderRegistry,
+    commit_tracker: Option<CommitTracker>,
```

Comment:
I like the idea of using write operations to notice that commit is needed, but having an "optional commit tracker" makes it confusing, IMO.

Instead, I would make this `commit_tracker: CommitTracker, and add `had_writes: bool` in `CommitTracker` itself.